src/julia-syntax.scm

```diff
@@ -1542,6 +1542,15 @@
         (cadr expr)  ;; eta reduce `x->f(x)` => `f`
         `(-> ,argname (block ,@splat ,expr)))))
 
+(define (ref-to-view expr)
+  (if (and (pair? expr) (eq? (car expr) 'ref))
+      (let* ((ex (partially-expand-ref expr))
+             (stmts (butlast (cdr ex)))
+             (refex (last    (cdr ex)))
+             (nuref `(call view ,(caddr refex) ,@(cdddr refex))))
```

Comment:
Yes; I would go with `top` to avoid capturing locals that might be called `view` or `broadcast`.